src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -645,9 +645,8 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return unchecked(value.GetHashCode() ^
-                    state.GetHashCode() ^
-                    knownColor.GetHashCode());
+            int hash = value.GetHashCode();
+            return unchecked((hash << 5) - hash + (knownColor.GetHashCode() << 3) + state);
```

Comment:
How about using:
https://github.com/dotnet/corefx/blob/master/src/Common/src/System/Numerics/Hashing/HashHelpers.cs
to combine them instead?